pydicom/util/dump.py

```diff
@@ -1,72 +1,106 @@
-# Copyright 2008-2018 pydicom authors. See LICENSE file for details.
+# Copyright 2008-2021 pydicom authors. See LICENSE file for details.
 """Utility functions used in debugging writing and reading"""
 
 from io import BytesIO
+import os
+import sys
+from typing import Union, Optional, BinaryIO, TYPE_CHECKING
 
+if TYPE_CHECKING:  # pragma: no cover
+    from pydicom.dataset import Dataset
 
-def print_character(ordchr):
+
+def print_character(ordchr: int) -> str:
     """Return a printable character, or '.' for non-printable ones."""
     if 31 < ordchr < 126 and ordchr != 92:
         return chr(ordchr)
-    else:
-        return '.'
+
+    return "."
 
 
-def filedump(filename, start_address=0, stop_address=None):
-    """Dump out the contents of a file to a
-       standard hex dump 16 bytes wide"""
+def filedump(
+    filename: Union[str, os.PathLike],
```

Comment:
`str` is already contained in `os.PathLike`.

Reply:
That's what I thought, but
```python
import os

value: os.PathLike
value = "foo"
```
Gives 

> error: Incompatible types in assignment (expression has type "str", variable has type "PathLike[Any]")  [assignment]

Reply:
PEP 519 has:

> This also removed any desire to have a class in the typing module which represented the union of all acceptable path-representing types as that can be represented with typing.Union[str, bytes, os.PathLike] easily enough and the hope is users will slowly gravitate to path objects only.

Although I don't really understand why `bytes` is included (2.7 compatibility?)

Reply:
Ok, I wasn't aware of this - when the documentation talked about `PathLike`, there was always `str` and `bytes` assumed alongside `pathlib.Path`. Obviously, this hasn't been extended to `typing`. Though turns out I just forgot this - I actually also had used an alias `AnyPath = Union[AnyStr, os.PathLike]` recently - bad memory...